helm-charts/opentelemetry-collector/values.yaml

```diff
@@ -0,0 +1,111 @@
+# Default values for opentelemetry-collector.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+nameOverride: ""
+fullnameOverride: ""
+
+# Basic parameters that used to compose configmaps
+telemetry:
+  metrics:
+    enabled: true
+  traces:
+    enabled: true
+output:
+  enabled: true
+  type: logging
+  logLevel: info
+
+# Shared params for agentCollector daemonset and standaloneCollector deployment pods.
+# Can be overriden here or for any component independantly.
+image:
+  repository: otel/opentelemetry-collector-contrib
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: ""
+imagePullSecrets: []
+
+# OpenTelemetry Collector executable
+command:
+  name: otelcontribcol
+  extraArgs: []
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+
+podAnnotations: {}
+
+podSecurityContext: {}
+securityContext: {}
+
+nodeSelector: {}
+tolerations: []
+affinity: {}
+
+# Configuration for ports, shared between agentCollector, standaloneCollector and service.
+# Can be overriden here or for agentCollector and standaloneCollector independently.
+# Any port can be disabled by setting to null.
+ports:
+  otlp:
+    containerPort: 55680
+    servicePort: 55680
+    hostPort: 55680
+    protocol: TCP
+  jaeger-thrift:
+    containerPort: 14268
+    servicePort: 14268
+    hostPort: 14268
+    protocol: TCP
+  jaeger-grpc:
+    containerPort: 14250
+    servicePort: 14250
+    hostPort: 14250
+    protocol: TCP
+  zipkin:
+    containerPort: 9411
+    servicePort: 9411
+    hostPort: 9411
+    protocol: TCP
+
+# Configuration for agent OpenTelemetry Collector daemonset, enabled by default
+agentCollector:
+  enabled: false
+
+  resources:
+    limits:
+      cpu: 256m
+      memory: 512Mi
+
+  # Configuration override that will be merged into the agent collector default config
+  configOverride: {}
+
+# Configuration for standalone OpenTelemetry Collector deployment, disabled by default
+standaloneCollector:
+  enabled: true
```

Comment:
The way the chart values are laid out make it seem like if one wanted both collector types in their cluster they could achieve this via 1 helm release, which may very well be the intention in which case you it may be desirable to embed more options in addition to `resources` such as independent `serviceAccount`s, `securityContext`s and so on. However if the intention is that one helm release result in either a standalone deployment OR an agent daemonset being deployed then I would move `resources` to top level, and use a top level option such as `collectorType: agent|standalone` or `kind: daemonset/deployment`.  WDYT?

Reply:
The intention is to be able to deploy both at the same time. It's a pretty common use case to have traces flowing through agents and concentrated/batched in a standalone collector. The k8s example we have shows exactly that https://github.com/open-telemetry/opentelemetry-collector/blob/master/examples/k8s.yaml. Not sure if both need to be enabled by default. I'd rather have just deamonset enabled by default. 

Having it in one release make it easier to configure traces/metrics flowing through the standalone collector by default it that one is enabled. That is achieved using this config override https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1026/files#diff-88a82221bc3c6b6f6ee0d5d04617e12e. And it involves less manual steps. So I'd prefer using one helm release over two.

I agree that it has to be flexible enough so every part (agent or standalone collector) gets it's own independent serviceAccounts, securityContexts etc. I'll define them once here at top level, but will make it possible to override for each independently.

Please let me know what do you think

CC @open-telemetry/collector-approvers

Reply:
ok sgtm good to know one release is meant to deploy both standalone and agent services

Reply:
I've just realized that I already made all the options like serviceAccount etc overridable for each of the component using this trick:  https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1026/files#diff-03f976ab7d04d60820f32aa59b25577eR24-R26 I needed agent/standalone specific options along with top level options in the pod template. It might be not very obvious, I'll put more descriptions in the values

Reply:
👏